utilityFunctions.f90

```diff
@@ -28,35 +32,75 @@ subroutine atmosphere( m, o2, n2 )
     return
   end subroutine atmosphere
 
-  subroutine zenith( ttime, dec )
+  ! ----------------------------------------------------------------- !
+
+  pure function calcDec( t ) result ( dec )
+    ! calculate the Sun Declination (radians): the sun declination is
+    ! the angle between the Sun and Earth's equatorial plane
+    use types_mod
+    use date, only : secondsInYear, dayAsFractionOfYear
+    implicit none
+
+    real(kind=DP), intent(in) :: t
+    real(kind=DP) :: dec, pi, daysInYear, maxDecInRad, currentFYear, temporary
+
+    daysInYear = 365.24
+    pi = 4.0 * atan( 1.0 )
+    maxDecInRad = 23.44 * pi / 180.0
+    currentFYear = dayAsFractionOfYear + ( t / secondsInYear )
+    temporary = 2.0 * 0.0167 * sin( 2 * pi * ( currentFYear - ( 2.0 / daysInYear ) ) )
+    dec = asin( sin( -maxDecInRad ) * cos( ( 2.0 * pi ) * ( currentFYear + ( 10.0 / daysInYear ) ) + temporary ) )
+    return
+  end function calcDec  ! NB: is this in radians or degrees?
+
+  ! ----------------------------------------------------------------- !
+
+  subroutine calcZenith( ttime, dec )
+    ! calculate the Local Hour Angle (radians) and the Solar Zenith
+    ! Angle (radians)
+    use types_mod
     use zenithData
     use SZACalcVars
     implicit none
 
     real(kind=DP), intent(in) :: ttime, dec
     real(kind=DP) :: pi, radian, rampValue
-    ! solar declination angle from July 1st - Harwell traj model
 
+    ! latitude and longitude (degrees)
     lat = latitude
     longt = longitude
+
+    ! calculate pi and deg->rad conversion factor
     pi = 4.0 * atan( 1.0 )
-    ! local hour angle - representing time of day
-    lha = ( 1.0 + ( ttime / 4.32D+04 ) ) * pi
     radian = 180.0 / pi
+
+    ! calculate local hour angle - representing time of day
+    ! LHA is the angle between observer's meridian and Sun's meridian
+    lha = ( 1.0 + ( ttime / 4.32D+04 ) ) * pi
+
+    ! convert latitude and sun declination to radians
+    !dec = dec/radian
     lat = lat / radian
-    ! dec = dec/radian
+
+    ! calculate cosine of solar zenith angle (cosx)
+    ! SZA is the angle between local vertical and center of the Sun
     sinld = sin( lat ) * sin( dec )
     cosld = cos( lat ) * cos( dec )
+    cosx = cos( lha ) * cosld + sinld
 
-    rampValue = ramp( cos( lha ) * cosld + sinld )
+    ! set negative cosx to zero and calculate the inverse (secx)
+    ! secx is needed for the MCM photolysis parameterisation
+    rampValue = ramp( cosx )
```

Comment:
Actually, do we really need this as a separate function? I believe it is used only here. What if I just substitute lines 93-94 with:

    ramped_cosx = max(0, cosx )
    secx = 1.0 / ( rampValue + 1.00d-30 )